akka-persistence/src/main/scala/akka/persistence/PersistentActor.scala

```diff
@@ -264,6 +264,28 @@ trait PersistentActor extends Eventsourced with PersistenceIdentity {
   def deferAsync[A](event: A)(handler: A ⇒ Unit): Unit = {
     internalDeferAsync(event)(handler)
   }
+
+  /**
+   * Defer the handler execution until all pending handlers have been executed. It is guaranteed that no new commands
+   * will be received by a persistent actor between a call to `defer` and the execution of its `handler`.
+   * Allows to define logic within the actor, which will respect the invocation-order-guarantee
+   * in respect to `persistAsync` or `persist` calls. That is, if `persistAsync` or `persist` was invoked before `defer`,
+   * the corresponding handlers will be invoked in the same order as they were registered in.
+   *
+   * This call will NOT result in `event` being persisted, use `persist` or `persistAsync` instead
+   * if the given event should possible to replay.
+   *
+   * If there are no pending persist handler calls, the handler will be called immediately.
+   *
+   * If persistence of an earlier event fails, the persistent actor will stop, and the `handler`
+   * will not be run.
+   *
+   * @param event event to be handled in the future, when preceding persist operations have been processes
+   * @param handler handler for the given `event`
+   */
+  def defer[A](event: A)(handler: A ⇒ Unit): Unit = {
+    internalDefer(event)(handler)
+  }
```

Comment:
since this is a trait the binary compatibility check will complain but as far as I can see it should be ok to add such method since we don't call it, and therefore it doesn't matter if it's not mixed in to users'/librararies' concrete class if that was compiled with older version

Reply:
I added exceptions to mima filters.